Source/WebCore/dom/ContainerNode.cpp

```diff
@@ -659,7 +659,7 @@ void ContainerNode::disconnectDescendantFrames()
     disconnectSubframesIfNeeded(*this, RootAndDescendants);
 }
 
-ExceptionOr<void> ContainerNode::removeChild(Node& oldChild)
+ExceptionOr<void> ContainerNode::removeChild(const Ref<Node> oldChild)
```

Comment:
cc @rniwa here but I thought we didn't do this, even with our new safer patterns.

Also, if we decide to do this, we should probably pass a `const Ref<Node>&` instead of a `const Ref<Node>`.

But maybe we should simply have a protector for oldChild inside the implementation?

Reply:
I see that I am contradicting Geoff above. However, passing a `const Ref<Node>&` or a `const Ref<Node>` is just not something we usually do in WebKit as far as I know.

I am curious if this is a new style we want and why we can't use a protector variable like we usually do.

Reply:
@cdumez is right that we don't usually do it this way. Our long term plan for local RefPtr verification would instead verify that the _caller_ held a Ref<Node> or RefPtr<Node> in scope.

Using a protector variable would also work here. An earlier version @achristensen07 showed me did it that way. 

My main feedback is that `const Ref<Node>&` is not completely safe.

If Node& argument and a protector variable are more congruent with other code, I'm cool with that too.

Reply:
I'm happy to do any of the 3 things which are mostly equivalent in this case if there's a consensus among reviewers

Reply:
OK, let's do local variable protector, to be similar to other code.